docker_build.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+if ! ( which docker ); then
+    echo "Docker not installed, unable to continue"
+    exit 1
+fi
+
+REPO_ROOT=$(pwd)
+PACKAGE_NAME="um-kernel"
+DOCKER_IMAGE_NAME="${PACKAGE_NAME}_build"
+DOCKER_CONTAINER_NAME="kernel_build"
+
+# Initialize repositories
+git submodule init
+git submodule update
+
+# Start clean
+sudo rm -rf ./_build_armhf 2> /dev/null
+sudo rm ./${PACKAGE_NAME}-*.deb 2> /dev/null
+
+if [[ -n $(docker ps -l | grep ${DOCKER_CONTAINER_NAME}) ]];then
+    echo "First remove the existing container"
+    docker rm -f ${DOCKER_CONTAINER_NAME} > /dev/null
+fi
+
+docker build -f Dockerfile . -t ${DOCKER_IMAGE_NAME}
+docker run --name ${DOCKER_CONTAINER_NAME} --privileged --cap-add=ALL -it -d -v ${REPO_ROOT}:/workspace ${DOCKER_IMAGE_NAME} bash
```

Comment:
I'm very nervous that we need priviledged and all capabilities for our very simple container!

And now I see the workspace requirement. Is this a jenkins requirement?

what jenkins would be doing is:
git clone locally
docker ....
(which docker can do in one go by saying docker build <file_to_git_Dockerfile>
which should pull in all the code files into the local context. At the very least, is that is how the local docker job works.

So what you are probably doing now, is adding the files to the container twice. Once via the local context copy, twice via the workspace volume.